components/input/textarea-count.component.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+import { AfterContentInit, ChangeDetectionStrategy, Component, ContentChild, ElementRef, Input, OnDestroy, Renderer2 } from '@angular/core';
+import { isNotNil } from 'ng-zorro-antd/core/util';
+import { NzInputDirective } from 'ng-zorro-antd/input/input.directive';
+import { EMPTY, Subject } from 'rxjs';
+import { map, startWith, takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'nz-textarea-count',
+  template: `
+    <ng-content select="textarea[nz-input]"></ng-content>
+  `,
+  host: {
+    class: 'ant-input-textarea-show-count'
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class NzTextareaCountComponent implements AfterContentInit, OnDestroy {
+  @ContentChild(NzInputDirective, { static: true }) nzInputDirective!: NzInputDirective;
+  @Input() nzMaxCharacterCount: number = 0;
```

Comment:
Changing either the maximum count or the compute function will not trigger the display to be updated.